core/lib/tasks/payment_method.rake

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+namespace :payment_method do
+  desc "Deprecates old payment methods and fixes ActiveRecord::SubclassNotFound error, "\
+  "which happens after switching Payment Service Provider."
+  task deprecate_unsupported_payment_methods: :environment do
+    unsupported_ids = get_unsupported_payment_method_ids
+
+    set_unsupported_payment_methods_type(unsupported_ids)
+    # Let ActiveRecord handle boolean format, which may vary
+    # depending on the DBMS in use.
+    set_unsupported_payment_methods_state(unsupported_ids)
+  end
+
+  def set_unsupported_payment_methods_type(unsupported_ids)
+    where_clause = get_where_clause('id', unsupported_ids)
+    ActiveRecord::Base.connection.execute(
+      "UPDATE spree_payment_methods "\
+      "SET type='#{Spree::PaymentMethod.name}' "\
```

Comment:
Why use `.name` instead of a string? Not saying it's wrong, I just would like to understand the reasoning behind it.

Reply:
I usually prefer to reference constants or variables when possible, just in case they may change in the future (even tough `Spree::PaymentMethod` is very unlikely to be renamed). Even having the tests, being able to reference a class to fix an error that requires that field to contain a valid class name seemed a bit safer to me.